src/validate.ts

```diff
@@ -159,25 +160,44 @@ export class CertiMintValidation {
             anchor.merkleRoot
           );
 
-          isValid = isValid && anchor.exists && validProof;
+          isValid = anchor.exists && validProof;
         }
       } else {
         for (const networkName of Object.keys(anchors[protocol])) {
-          console.log(anchors[protocol]);
-          console.log('succcesfully anchored!');
-          // bitcoinJs.Transaction.fromHex(anchor.transactionId);
-          isValid = true;
-          console.log('It works till here');
-
-          console.log('But not till here');
           const anchor = anchors[protocol][networkName];
           try {
             const txId = anchor.transactionId;
             const tx = await Axios.get(this.buildTxUrl(baseUrl, txId));
-            console.log(anchor, tx, 'ello');
-            console.log(anchor, networkName);
+
+            const txOutputs = tx.data.outputs;
+            let merkleRoot: string;
+            for (const output of txOutputs) {
+              if (output.data_hex !== undefined && output.data_hex != null) {
+                merkleRoot = output.data_hex;
+              }
+            }
+
+            anchor.exists = merkleRoot === anchor.merkleRoot;
+
+            const merkleTools = new MerkleTools({
+              hashType: 'SHA3-512'
+            });
+
+            const validProof = merkleTools.validateProof(
+              anchor.proof,
+              dataHash,
+              anchor.merkleRoot
+            );
+
+            isValid = anchor.exists && validProof;
           } catch (error) {
-            console.log(error);
+            if (error.response !== undefined && error.response.status === 429) {
+              throw new Error(
+                'Too many request to the blockcypher api, please add an apikey or upgrade your blockcypher plan'
+              );
+            } else {
+              throw error;
```

Comment:
REQ: watch out - you changed logic here again - we used to fail and silently and log. Unsure what to do here, depends on who's calling and how that's being handled. I think yours is better but we need to be sure.

Reply:
Q: So should I remove this code then or should I rewrite all the error catching in general from this repo?